### PR TITLE
dev-docs: document patching conventions

### PR DESCRIPTION
**Pull request by katexochen**

### CODE REVIEW

File: dev-docs/patches.md [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+  (fetchpatch {
+    url = "https://github.com/...";
+    stripLen = 2; # Sometimes the patch strip level must be configured
+    hash = "sha256-AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA=";
+  })
+];
+```
+
+If there aren't any changes needed to an upstream patch, it's good practice to fetch the patch from remote. If the changes to upstream patches are needed or patches aren't available upstream, the
+*full patch set* should be vendored, so it can easily be worked with.
```
Comment (burgerdev) [inline dev-docs/patches.md:52]: Can you define "full patch set" here?

File: dev-docs/patches.md [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+patches = [
+  (fetchpatch {
+    url = "https://github.com/...";
+    stripLen = 2; # Sometimes the patch strip level must be configured
+    hash = "sha256-AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA=";
+  })
+];
+```
+
+If there aren't any changes needed to an upstream patch, it's good practice to fetch the patch from remote. If the changes to upstream patches are needed or patches aren't available upstream, the
```
Comment (burgerdev) [inline dev-docs/patches.md:51]: I'd argue that a patch needs to be in some reliable branch (main, release/xxx) to qualify for remote inclusion, to minimize the chance of losing it due to garbage collection etc.

File: dev-docs/patches.md [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+
+When done, recreate the patch set:
+
+```sh
+git format-patch -o $pkgDir $rev
+```
+
+# Patch documentation conventions
+
+Patches need thorough documentation. Each reference of a patch must have a comment.
```
Comment (burgerdev) [inline dev-docs/patches.md:88]: > Each reference of a patch must have a comment.

You mean a comment in the `.patch` file, after the `---`?

**Review by burgerdev — APPROVED** [review]

File: dev-docs/patches.md (hunk 2 above)
Comment (katexochen) [reply dev-docs/patches.md:51]: While I see the point of garbage collection on the GitHub side, I don't think this is too big of a problem. From my experience they don't run aggressive gc on repos and commit stay around for quite some time. The reason I'd rather not restrict this further is that it is quite common to reference a commit from an open PR that we've just opened, which is ok imo.

Actually did some research, seems like at least in some scenarios dangling commits are kept around for quite some time: https://stackoverflow.com/questions/4367977/how-to-remove-a-dangling-commit-from-github

File: dev-docs/patches.md (hunk 1 above)
Comment (katexochen) [reply dev-docs/patches.md]: tried to clarify this

File: dev-docs/patches.md (hunk 3 above)
Comment (katexochen) [reply dev-docs/patches.md]: I've added an example and some reasoning about it.

File: dev-docs/patches.md (hunk 1 above)
Comment (burgerdev) [reply dev-docs/patches.md]: Clear now, thanks!

File: dev-docs/patches.md (hunk 2 above)
Comment (burgerdev) [reply dev-docs/patches.md]: I like how it's phrased now, and thanks for doing the research.

**Review by burgerdev — APPROVED** [review]

**Pull request merged**